benchmarks/decoders/benchmark_decoders_library.py

```diff
@@ -299,6 +301,55 @@ def create_torchcodec_decoder_from_file(video_file):
     return video_decoder
 
 
+def generate_video(command):
+    print(command)
+    print(" ".join(command))
+    subprocess.check_call(command)
+
+
+def generate_videos(
+    resolutions,
+    encodings,
+    fpses,
+    gop_sizes,
+    durations,
+    pix_fmts,
+    ffmpeg_cli,
+    output_dir,
+):
+    executor = ThreadPoolExecutor(max_workers=20)
+    video_count = 0
+    for resolution in resolutions:
+        for duration in durations:
+            for fps in fpses:
+                for gop_size in gop_sizes:
+                    for encoding in encodings:
+                        for pix_fmt in pix_fmts:
+                            outfile = f"{output_dir}/{resolution}_{duration}s_{fps}fps_{gop_size}gop_{encoding}_{pix_fmt}.mp4"
+                            command = [
+                                ffmpeg_cli,
+                                "-y",
+                                "-f",
+                                "lavfi",
+                                "-i",
+                                f"color=c=blue:s={resolution}:d={duration}",
+                                "-c:v",
+                                encoding,
+                                "-r",
+                                f"{fps}",
+                                "-g",
+                                f"{gop_size}",
+                                "-pix_fmt",
+                                pix_fmt,
+                                outfile,
+                            ]
+                            executor.submit(generate_video, command)
+                            video_count += 1
+
+    executor.shutdown(wait=True)
+    print(f"Generated {video_count} videos")
+
+
```

Comment:
On the plot below the "decoder" legend is repeated: on the left, and on every single plot. It might be best to only keep one so as not to overload the plots

Reply:
Agreed it would look best with a single legend outside of all plots. I didn't say anything before because I wasn't sure how, but digging through old matplotlib code from an old research project, I think I found an example: [experiments/process_bulk_evict.py](https://github.com/IBM/sliding-window-aggregators/blob/566b7222c205de242b87ce65013949c35e518a8b/experiments/process_bulk_evict.py#L144). We have to create subplots, and put the legend on the graph object, not the axes.